sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamEnumerableConverter.java

```diff
@@ -141,19 +141,25 @@ public static PipelineOptions createPipelineOptions(Map<String, String> map) {
   }
 
   static List<Row> toRowList(PipelineOptions options, BeamRelNode node) {
+    checkAndFailOnNonDirectRunner(options, "toRowList is only available in DirectRunner.");
+
     if (node instanceof BeamIOSinkRel) {
       throw new UnsupportedOperationException("Does not support BeamIOSinkRel in toRowList.");
```

Comment:
Can we support count here? I.e. if we make `count()` return a singleton `Collection<Row>` with a single field and element?

Reply:
There is no direct usage on it: people when want access the count can call `toEnumerable`. 

`toRowList` is used for cases when plain Row is required. 

Count, instead, is an int without associated schema.

Reply:
I am thinking this would allow to reduce the code duplication, see my other comment.

All relational operations though have a schema. My understanding is that when I'm running `SELECT 1` I still get a relation as the result. Same for `SELECT COUNT(x)`.

Now `count()` here is an int without an associated schema in the current implementation which doesn't allow to use it in `toRowList()` as you mentioned. However if you wrap the value in a row with a schema, it probably should work. And even if it's not used explicitly it might allow to re-use the code between `toRowList` and `toEnumerable`.

And in `toEnumerable`, the call to `rowToAvaticaAndUnboxValues` should handle the single row with a single value as well, so that you don't have to special case it there.

Reply:
Ah this is the last comment I saw, but it explains why there is a propose to merge count for both API.

I wouldn't call this proposal as a way to reduce code duplication: there is no duplicated code now because toRowList does not support count()

Reply:
Basically if there is a need to support `count` for `toRowList`, it's better to use the same `count` for both API. 

It's just debatable that if `count` is needed for `toRowList`.

Reply:
It's not needed. But we are in a situation like this:
 * it is not needed, so we have a special case for it which throws an exception, and if we need it we will have to implement it;
 * it's free to add support for it (-ish, modulo wrapping/unwrapping the value into a row) which remove special casing and make it simpler;